src/pt/huntersscans/src/eu/kanade/tachiyomi/extension/pt/huntersscans/HuntersScanDescrambler.kt

```diff
@@ -0,0 +1,66 @@
+package eu.kanade.tachiyomi.extension.pt.huntersscans
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.util.Base64
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.InputStream
+import java.nio.charset.Charset
+
+object HuntersScanDescrambler {
+
+    fun decryptHuntersPayload(payloadB64: String, keyB64: String): List<String> {
+        val payload = String(Base64.decode(payloadB64, Base64.DEFAULT), Charset.forName("ISO-8859-1"))
+        val key = String(Base64.decode(keyB64, Base64.DEFAULT), Charset.forName("ISO-8859-1"))
+        val sb = StringBuilder()
+
+        for (i in payload.indices) {
+            var keyIndex = (i % key.length) - 1
+            if (keyIndex < 0) keyIndex = key.length + keyIndex
+            val c = payload[i].code - key[keyIndex].code
+            sb.append(c.toChar())
+        }
```

Comment:
Consider using `forEachIndexed` instead, and replacing `var` with `val`.

```suggestion
        payload.forEachIndexed { i, ch ->
            val keyIndex = (i + key.length - 1) % key.length
            val c = ch.code - key[keyIndex].code
            sb.append(c.toChar())
        }
```